src/languages/nim/emit.ts

```diff
@@ -22,51 +21,40 @@ function emitMultiExpr(expr: IR.Expr, parent: IR.Node): TokenTree {
       "\n"
     );
   }
-  if (containsMultiExpr(children)) {
-    return [
-      "$INDENT$",
-      "\n",
-      joinTrees(
-        children.map((stmt) => emitStatement(stmt, expr)),
-        "\n"
-      ),
-      "$DEDENT$",
-    ];
+  let inner = [];
+  let needsBlock = false;
+  for (const child of children) {
+    const needsNewline =
+      "consequent" in child || "children" in child || "body" in child;
```

Comment:
Compare with `containsMultiExpr` from common/emit.ts

```ts
export function containsMultiExpr(exprs: readonly IR.Expr[]): boolean {
  for (const expr of exprs) {
    if ("consequent" in expr || "children" in expr || "body" in expr) {
      return true;
    }
  }
  return false;
}
```

These could have a common helper function, in case we ever add a node with a new multi-expr.

Reply:
> Compare with `containsMultiExpr` from common/emit.ts
> 
> ```ts
> export function containsMultiExpr(exprs: readonly IR.Expr[]): boolean {
>   for (const expr of exprs) {
>     if ("consequent" in expr || "children" in expr || "body" in expr) {
>       return true;
>     }
>   }
>   return false;
> }
> ```
> 
> These could have a common helper function, in case we ever add a node with a new multi-expr.

I think the problem is, that at a Polygolf level the distinction is missing and language targets can disagree on what needs a block and what doesn't.